### ISSUE TITLE
Automatically Approve or Deny Requests based on Studio/Network or even streaming service availability

### ISSUE THREAD

**Issue by drpeppershaker** [sct/overseerr#835]
**Is your feature request related to a problem? Please describe.**
I know all my plex users have a Netflix account. I do not necessarily want them requesting things that are already available on Netflix. My server space is limited, and I do not wish to download things that are readily available to my users. Currently I deny those requests manually.

**Describe the solution you'd like**
I would like to see a way--even if it's a hidden/advanced setting, to automatically deny any request for Netflix content. I understand that maintaining a database of currently available content on various streaming platforms is a lofty and likely impossible/impractical goal. Therefore, I propose using the Network/Studio information already available as a way automatically deny requests from a specific studio/network.

**Describe alternatives you've considered**
I've considered making a local user account to request every single Netflix original account and then logging in with my admin account to deny all them to prevent them from being requested again, but I don't know if that would work, and it seems like a pretty big undertaking to do it all manually.

**Additional context**
This could also be beneficial for those looking to implement parental controls, or other type of controls on their requests as well.

Also, If it is at all useful, it appears that https://github.com/andreluisos/lists currently maintains a list of content on various streaming services that could be beneficial to this request -- if using tmdb network/studio info isn't as helpful as I thought.

**Comment by costaht** [comment]
It would be simpler if they just hid the content from netflix

**Comment by TheCatLady** [comment]
Netflix has a lot of content that it doesn't produce itself, however.  I'm not sure that checking the studio/network would solve your issue here, unless you only want to block Netflix Originals?

I think in general, some global content filtering might be nice though.  I don't like the idea of depending on a non-TMDb list, but I'm all for filtering based on existing TMDb data.

**Comment by drpeppershaker** [comment]
> Netflix has a lot of content that it doesn't produce itself, however. I'm not sure that checking the studio/network would solve your issue here, unless you only want to block Netflix Originals?
> 
> I think in general, some global content filtering might be nice though. I don't like the idea of depending on a non-TMDb list, but I'm all for filtering based on existing TMDb data.

Pie in the sky big dream would be to track which services have which content--but not realistic .Non-original content is constantly changing hands between various platforms and would likely be near impossible to track.  Plus, you'd have to deal with some content being available in some geo-locations, but not others, no API access (that I'm aware of), etc...
I think being able to block the originals would be the simplest way to implement a streaming service filter. 

I guess, more generally this request is: can we get some global filters if we choose to implement them, and can studio/network be one of the filters? Solves my problem (prevent users from requesting readily available content), adds a nice feature to Overseerr for other use-cases as well.

**Comment by costaht** [comment]
> Netflix has a lot of content that it doesn't produce itself, however. 

Stupid me... I didn't think about that. Yeah, blocking the originals of the provider seems the better way to go.

**Comment by Reupenny** [comment]
If this is ever possible it would be nice to have it show “play on Netflix,Disney+,ect” like it does with content on plex.
I know that justwatch.com maintains a pretty expansive data base on what looks to be every streaming service.

**Comment by 2fst4u** [comment]
JustWatch provides an API that can be used for exactly this. I was going to put in a request for this not long ago but you beat me to it!

JustWatch have geo-specific availability so you could have a request go through the API, confirm it isn't available on the streaming services set, and then deny or let through based on that decision.

**Comment by mouzzampk2014** [comment]
> Netflix has a lot of content that it doesn't produce itself, however. I'm not sure that checking the studio/network would solve your issue here, unless you only want to block Netflix Originals?
> 
> I think in general, some global content filtering might be nice though. I don't like the idea of depending on a non-TMDb list, but I'm all for filtering based on existing TMDb data.

Yup thats the one and i also think filtering will give more options like filter by release date or genres

**Comment by 2fst4u** [comment]
This doesn't need to be tied to studio. It only needs to check for availability by service. That's where the JustWatch API will help.

**Comment by sct** [comment]
> This doesn't need to be tied to studio. It only needs to check for availability by service. That's where the JustWatch API will help.

As far as I know JustWatch doesn't have a public API?

**Comment by 2fst4u** [comment]
It may not be fully open and public but is this not what would be required?

https://www.justwatch.com/nz/JustWatch-Streaming-API

**Comment by 2fst4u** [comment]
Bump

**Comment by 2fst4u** [comment]
Bump

**Comment by tohn** [comment]
Maybe this is similar to #654? And there is already a PR for this: #1778 🎉

**Comment by 2fst4u** [comment]
Even if similar, which is doesn't sound entirely the same as, this issue requests the functionality of automatically approving or denying based on the logic.

**Comment by LeaderbotX400** [comment]
bumb

**Comment by 2fst4u** [comment]
Now that Overseerr shows where an item is currently streaming it seems the hard part is done. We just need a setting to filter these streaming platforms to automatically deny if desired.

**Comment by epheterson** [comment]
Bumping again, would be great to have! We already have “streaming on” so not much more to add. 

Some ideas:
- Make “streaming on” section logos instead of text to make it more obvious
- Instead of denying, remove the button / ability to request
- Where feasible, add direct URLs to streaming services to easily start playback

These can help the user steer clear of downloading something they can already easily access, but may not realize, instead of making a request and denying.

**Comment by TheCatLady** [comment]
> * Make “streaming on” section logos instead of text to make it more obvious

Logos are not guaranteed to be returned by the API. (Same issue for networks / production companies.)

> * Where feasible, add direct URLs to streaming services to easily start playback

Not possible, we don't have the data required to add deeplinks.

**Comment by epheterson** [comment]
> > * Make “streaming on” section logos instead of text to make it more obvious
> 
> Logos are not guaranteed to be returned by the API. (Same issue for networks / production companies.)

Many logos already appear correctly in the main browsing view as tiles, could be nice to show logos when available. 

> > * Where feasible, add direct URLs to streaming services to easily start playback
> 
> Not possible, we don't have the data required to add deeplinks.

Looks like it might be out there for some services, definitely more a nice to have than necessary feature though. 
https://reelgood.com/business/products/catalog/
https://rapidapi.com/blog/movie-api/

Just some ideas! Thanks for considering.

**Comment by evilhyde** [comment]
Bump

From mobile
________________________________

**Comment by 2fst4u** [comment]
Bump

**Comment by LeaderbotX400** [comment]
Bump

**Comment by evilhyde** [comment]
Bump
________________________________

**Comment by ManlyJack** [comment]
Bump, This needs to happen

**Comment by kingandaval** [comment]
> Bumping again, would be great to have! We already have “streaming on” so not much more to add.
> 
> Some ideas:
> 
> * Make “streaming on” section logos instead of text to make it more obvious
> * Instead of denying, remove the button / ability to request
> * Where feasible, add direct URLs to streaming services to easily start playback
> 
> These can help the user avoid requesting easily accessible content, instead of them making a request and denying.

Super new to Overseerr... am I missing something with 'streaming on' as I can't see that at all?

**Comment by epheterson** [comment]
Yeah go to a show or movie that’s streaming somewhere and there’s a section, “Currently Streaming On.”

**Comment by 2fst4u** [comment]
I think you have to set your region for that feature to work, but that presents another issue that only movies and shows from your region will show on the main page which is not desired.

**Comment by etfeet** [comment]
I'd been interested in this. not for this particular use case by for having filters to determine when/if to auto approve a request and what sonarr/radarr instance that media request gets sent to. An example of an app that does that fairly well is Petio. which allows you to create rules for just about any sonarr/radarr metadata field and additionally use those fields to determine which sonarr/radarr instance the request gets routed too. so it would also allow say anime movies going to a dedicated anime radarr instance.

![image](https://user-images.githubusercontent.com/11986383/219935294-2f0291c7-e216-4306-ac14-172f9133ebc4.png)

**Comment by JohnnyGrey86** [comment]
I'd like to toss my vote in for this as well. Ever since the "Currently Streaming On" feature was added, I think it would be cool if each user were presented the option to choose what services they subscribe to, then have Overseerr automatically block those requests. Or as mentioned above, hide/disable the button.

**Comment by nicobleiler** [comment]
Maybe even have a play on Netflix, Amazon or whatever button, similar to the one if it is on Plex

**Comment by drpeppershaker** [comment]
Bump

**Comment by kingandaval** [comment]
> Bump

Was just about to do the same :)

**Comment by 2fst4u** [comment]
Bump

**Comment by drpeppershaker** [comment]
Bump Sent from my iPhoneOn Sep 16, 2023, at 4:13 PM, stale[bot] ***@***.***> wrote:﻿
This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.

**Comment by Niborian** [comment]
I would also like to have a way to auto deny netflix requests by using the currently streaming on... if this could be combined with something to make it more accurate (geolocation) then that would be even better, but it would be nice to be able to just auto deny or temporary deny, so I can override if its not avilable in my country... or just confirm deny if it is...

**Comment by Alsidsds** [comment]
This would be very nice because I sometimes want to add movies/tv shows on plex's watchlist just so I don't forget about them but they are available on streaming platforms. I also don't want to disable auto requests on overseerr from plex's watchlist because it's very useful so at the moment I just don't add shows/movies available on streaming on the watchlist so they don't get requested on overseerr but it kinda defeats the purpose of the watchlist on plex being a hub for everything you want to watch wherever they are available.

**Comment by stevooo2** [comment]
For me, a helpful interim solution would be listing the streaming services a movie/show is available on (based on users country) within the movie/show description.  This would be enough for users to at least think twice before they hit the request button.

Ombi works this way today - see attached screenshot as an example from the [Ombi demo page](https://app.ombi.io/).

![ombi](https://github.com/user-attachments/assets/7dc21069-1380-4a01-aeea-265db98680f3)

**Comment by WEIRDesign** [comment]
> For me, a helpful interim solution would be listing the streaming services a movie/show is available on (based on users country) within the movie/show description. This would be enough for users to at least think twice before they hit the request button.
> 
> Ombi works this way today - see attached screenshot as an example from the [Ombi demo page](https://app.ombi.io/).
> 
> ![ombi](https://private-user-images.githubusercontent.com/55281407/355562933-7dc21069-1380-4a01-aeea-265db98680f3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkwODQ0NTgsIm5iZiI6MTcyOTA4NDE1OCwicGF0aCI6Ii81NTI4MTQwNy8zNTU1NjI5MzMtN2RjMjEwNjktMTM4MC00YTAxLWFlZWEtMjY1ZGI5ODY4MGYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE2VDEzMDkxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFlNjcxODAyM2YwNTM0NDg2MDg2NmU3YmM2ZmY2N2IzNDgyODkzM2NkNzYwMDQ2NjI1YjhkZWE4YWY1YmE4OTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.2EBkU2Eo-rTFfD-nWpUGGh7xpc1mw5JBxjtDTWoRSmg)

Is this not already a feature? (Albeit text only, and SkyGo isn't showing for me on Deadpool, which could be down to timing as I'm replying 2 months later! :D)

![image](https://github.com/user-attachments/assets/e875dbd2-7890-4368-abcc-4604dfdaa5b8)

**Comment by conorlap** [comment]
Bump

**Comment by carnivorouz** [comment]
bump

**Comment by RChesterton** [comment]
Would it be possible to allow us to set more than one "Discover Region" in settings, so it could list, for example, all streaming services in the USA and UK on the item page?

**Comment by Jopinder** [comment]
Not stale.

**Comment by KingAndaval** [comment]
> Not stale.

Overseerr is being merged with Jellyseerr and is now called Seerr with both dev teams joining together. 

https://github.com/seerr-team/seerr
